sdk/runtime/runner.go

```diff
@@ -20,11 +21,33 @@ func Invoke(tc func(*RunEnv) error) {
 	// The prometheus pushgateway has a customized scrape interval, which is used to hint to the
 	// prometheus operator at which interval the it should be scraped. This is currently set to 5s.
 	// To provide an updated metric in every scrape, jobs will push to the pushgateway at the same
-	// interval. When this "push_interval" is changed, you may want to change the scrape interval
+	// interval. When this "pushInterval" is changed, you may want to change the scrape interval
 	// on the pushgateway
 	pushStopCh := make(chan struct{})
 	go func() {
 		pushInterval := 5 * time.Second
+
+		// Wait until the pushgateway is ready.
+		runenv.RecordMessage("Waiting for pushgateway to become accessible.")
+		var resbuf []byte
+		for {
+			select {
+			case <-time.After(pushInterval):
+				resp, err := http.Get("http://prometheus-pushgateway:9091/-/ready")
```

Comment:
I *definitely* want to get rid of the hard-coded urls everywhere. Should turn this into an env variable, like the REDIS_HOST, I think.

Reply:
Yes, go for it.

* `cluster:k8s` and `local:docker` runners would inject `prometheus-pushgateway:9091`
* `local:exec` should inject `localhost:9091`.

Reply:
Oh, nice. I hadn't realised that pushgateway had a `ready` endpoint.

Reply:
How does this interact with the defer below, which actually forces a final push? We should probably capture the error there and make it less scary, with a nicer message like: `metrics push failed during instance shutdown; prometheus metrics from this instance may not be available; err: %s`

Reply:
it does print the final message " error during end metric push...", right now. But of course that could be changed.

Reply:
I created https://github.com/ipfs/testground/issues/711 to change this into a environment-delivered parameter.